src/channel.rs

```diff
@@ -129,6 +142,22 @@ impl ChannelBuilder {
         self
     }
 
+    /// Set resource quota by consuming a ResourceQuota
+    pub fn set_resource_quota(mut self, quota: ResourceQuota) -> ChannelBuilder {
+        let raw_quota = quota.get_ptr();
+        unsafe {
+            grpc_sys::grpc_resource_quota_ref(raw_quota);
+            self.options.insert(
+                Cow::Borrowed(grpc_sys::GRPC_ARG_RESOURCE_QUOTA),
+                Options::Pointer(
+                    raw_quota as _,
```

Comment:
Why not store it as type `ResourceQuota`? So that explicit `Drop` impl is not necessary.

Reply:
Good Point, `ResourceQuota` don't have to deal with destructuring